listenbrainz/webserver/static/js/src/stats/UserEntity.tsx

```diff
@@ -0,0 +1,452 @@
+        </div>
+        <div className="row">
+          <div className="col-xs-12">
+            <h3>
+              Top {entity.charAt(0).toUpperCase() + entity.slice(1)}s of{" "}
+              {range !== "all_time" ? "the" : ""}
+              <span className="dropdown">
+                <button
+                  className="dropdown-toggle btn-transparent"
+                  data-toggle="dropdown"
+                  type="button"
+                  style={{
+                    textTransform: "capitalize",
+                    fontWeight: "bold",
+                  }}
+                >
+                  {`${range.replace(/_/g, " ")}`}
+                  <span className="caret" />
+                </button>
+                <ul className="dropdown-menu" role="menu">
+                  <li>
+                    <a
+                      href="#"
+                      onClick={() => this.changeRange("week")}
+                      role="button"
+                    >
+                      Week
+                    </a>
+                  </li>
+                  <li>
+                    <a
+                      href="#"
+                      onClick={() => this.changeRange("month")}
+                      role="button"
+                    >
+                      Month
+                    </a>
+                  </li>
+                  <li>
+                    <a
+                      href="#"
+                      onClick={() => this.changeRange("year")}
+                      role="button"
+                    >
+                      Year
+                    </a>
+                  </li>
+                  <li>
+                    <a
+                      href="#"
+                      onClick={() => this.changeRange("all_time")}
+                      role="button"
+                    >
+                      All Time
+                    </a>
+                  </li>
+                </ul>
+              </span>
+              {range === "week"
+                ? `of ${startDate.getDate()} ${month[startDate.getMonth()]} `
+                : ""}
+              {range === "month" ? `${month[startDate.getMonth()]} ` : ""}
+              {range !== "all_time" ? startDate.getFullYear() : ""}
+            </h3>
+          </div>
+        </div>
+        <div className="row">
+          <div className="col-xs-12">
+            <h4 style={{ textTransform: "capitalize" }}>
+              {entity} count - <b>{entityCount}</b>
+            </h4>
+          </div>
+        </div>
+        <div className="row">
+          <div
+            className="col-md-12"
+            style={{ height: `${(75 / this.ROWS_PER_PAGE) * data.length}em` }}
+          >
+            <Bar data={data} maxValue={maxListens} />
+          </div>
+        </div>
+        <div className="row">
+          <div className="col-xs-12">
+            <ul className="pager">
+              <li className={`previous ${!(prevPage > 0) ? "hidden" : ""}`}>
```

Comment:
I usually prefer buttons to be visible but disabled, but it's personal preference.
Also, it would mean changing it in other pages with pagination, so maybe for another day.